extensions/amp-tiktok/validator-amp-tiktok.protoascii

```diff
@@ -0,0 +1,112 @@
+#
+# Copyright 2021 The AMP HTML Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the license.
+#
+
+tags: {  # amp-tiktok
+  html_format: AMP
+  tag_name: "SCRIPT"
+  extension_spec: {
+    name: "amp-tiktok"
+    version: "0.1"
+    version: "latest"
+  }
+  attr_lists: "common-extension-attrs"
+}
+tags: {  # <amp-tiktok>
+  html_format: AMP
+  tag_name: "AMP-TIKTOK"
+  spec_name:"AMP-TIKTOK video id"
+  requires_extension: "amp-tiktok"
+  attr_lists: "extended-amp-global"
+  attrs: {
+    name: 'data-src';
+    value_regex: '[0-9]+';
+  }
+  spec_url: "https://amp.dev/documentation/components/amp-tiktok"
+  amp_layout: {
+    supported_layouts: FILL
+    supported_layouts: FIXED
+    supported_layouts: FIXED_HEIGHT
+    supported_layouts: FLEX_ITEM
+    supported_layouts: INTRINSIC
+    supported_layouts: NODISPLAY
+    supported_layouts: RESPONSIVE
+  }
+}
+tags: {  # <amp-tiktok>
+  html_format: AMP
+  tag_name: "AMP-TIKTOK"
+  spec_name:"AMP-TIKTOK video source url"
+  requires_extension: "amp-tiktok"
+  attr_lists: "extended-amp-global"
+  attrs: {
+    name: 'data-src';
+    value_url: {
+      protocol: 'https';
+      allow_relative: false;
+      allow_empty: false;
+    }
+  }
+  spec_url: "https://amp.dev/documentation/components/amp-tiktok"
+  amp_layout: {
+    supported_layouts: FILL
+    supported_layouts: FIXED
+    supported_layouts: FIXED_HEIGHT
+    supported_layouts: FLEX_ITEM
+    supported_layouts: INTRINSIC
+    supported_layouts: NODISPLAY
+    supported_layouts: RESPONSIVE
+  }
+}
+tags: {  # <amp-tiktok>
+  html_format: AMP
+  tag_name: "AMP-TIKTOK"
+  spec_name:"AMP-TIKTOK blockquote"
+  requires_extension: "amp-tiktok"
+  attr_lists: "extended-amp-global"
+  spec_url: "https://amp.dev/documentation/components/amp-tiktok"
+  amp_layout: {
+    supported_layouts: FILL
+    supported_layouts: FIXED
+    supported_layouts: FIXED_HEIGHT
+    supported_layouts: FLEX_ITEM
+    supported_layouts: INTRINSIC
+    supported_layouts: NODISPLAY
+    supported_layouts: RESPONSIVE
+  }
+  reference_points: {
+    tag_spec_name: "AMP-TIKTOK blockquote [child]"
+  }
+}
+tags: {
+  html_format: AMP
+  tag_name: "$REFERENCE_POINT"
```

Comment:
Are you sure you need to use reference points for this? I think an alternative is:

1. In the tag_spec above, use `child_tags { child_tag_name_oneof: "blockquote"; mandatory_min_num_child_tags: 1 }`
2. In this tag spec, make it `tag_name: "blockquote"; mandatory_parent: "AMP-TIKTOK blockquote"`.

But don't trust me on all that; my validator-fu is weak. I generally lean away from reference points because they're confusing to me, but your call.

Reply:
I have updated the implementation to use the child_tag. In this case you were totally right, this is the right way to set it up. Thanks for the tip!

Reply:
Thanks!